src/App.js

```diff
@@ -14,11 +16,53 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGood: 'Jam',
+  };
+
+  render() {
+    const { selectedGood } = this.state;
+
+    return (
+      <>
+        {selectedGood ? (
+          <h1>
+            {`Selected good: - ${selectedGood} is selected`}
+            <button
+              type="button"
+              onClick={() => {
+                this.setState({ selectedGood: null });
+              }}
+            >
+              x
+            </button>
+          </h1>
+        ) : (
+          <h1>No value selected</h1>
+        )}
+
+        {goodsFromServer.map(good => (
+          <ul>
+            <li type="square">
```

Comment:
```suggestion
        <ul>
          {goodsFromServer.map(good => (
            <li key={good} type="square">
```
Now you are creating many lists `ul` with one list item `li`. You should create one list with many mapped items inside.
Also, the `key` attribute should be applied to `li`, not to `button`